### PR TITLE
Fix Nested objects and Arrays for Form Validation

### PR DESCRIPTION
**Pull request by giacomoaugello1**
- I fixed some errors that were coming up when running the "npm run lint" command, just to be able to commit.
- I made changes to the FormControl and Form components regarding how the "checkResult" object returned by the SchemaModel was parsed. It basically didn't take into account that there could be more than one level of nesting, and also didn't consider arrays.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -68,7 +68,8 @@
     "react-use-set": "^1.0.0",
     "react-window": "^1.8.8",
     "rsuite-table": "^5.18.2",
-    "schema-typed": "^2.1.3"
+    "schema-typed": "^2.1.3",
+    "typescript": "^4.4.3"
```
Comment (simonguo) [inline package.json:72]: typescript should not be a dependencies, but a devDependencies.

File: src/Accordion/Accordion.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
-import PanelGroup, { type PanelGroupProps } from '../PanelGroup';
+import PanelGroup from '../PanelGroup';
+import type { PanelGroupProps } from '../PanelGroup';
```
Comment (simonguo) [inline src/Accordion/Accordion.tsx:3]: Unrelated code changes to the topic

File: src/AutoComplete/Combobox.tsx [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { useCombobox } from '../internals/Picker';
-import Input, { type InputProps } from '../Input';
+import Input from '../Input';
+import type { InputProps } from '../Input';
```
Comment (simonguo) [inline src/AutoComplete/Combobox.tsx:4]: Unrelated code changes to the topic

File: src/DateInput/utils.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-import { type Locale } from 'date-fns';
+import type { Locale } from 'date-fns';
```
Comment (simonguo) [inline src/DateInput/utils.ts:1]: Unrelated code changes to the topic

File: src/DateRangeInput/utils.ts [hunk 5]
```diff
@@ -1,4 +1,4 @@
-import { type Locale } from 'date-fns';
+import type { Locale } from 'date-fns';
```
Comment (simonguo) [inline src/DateRangeInput/utils.ts:1]: Unrelated code changes to the topic

File: src/FormControl/FormControl.tsx [hunk 6]
```diff
@@ -162,8 +162,9 @@
 
   const getFieldError = (fieldName: string) => {
     if (nestedField) {
-      const name = fieldName.includes('.')
-        ? fieldName.replace('.', '.object.') + '.errorMessage'
+      const name = /[\.\[\]]+/g.test(fieldName)
+        ? fieldName.replace(/\./g, '.object.').replace(/\[([^\]]*)\]/g, '.array[$1]') +
```
[the remark this answers is not in the document]
Comment (simonguo) [reply src/FormControl/FormControl.tsx:166]: Polynomial regular expression used on uncontrolled data. [Show more details](https://github.com/rsuite/rsuite/security/code-scanning/6).
may run slow on strings starting with '[' and with many repetitions of '[\'.

File: src/internals/Picker/hooks/useCombobox.ts [hunk 7]
```diff
@@ -1,5 +1,6 @@
 import { useContext } from 'react';
-import { ComboboxContextContext, type ComboboxContextProps } from '../PickerToggleTrigger';
+import { ComboboxContextContext } from '../PickerToggleTrigger';
+import type { ComboboxContextProps } from '../PickerToggleTrigger';
```
Comment (simonguo) [inline src/internals/Picker/hooks/useCombobox.ts:3]: Unrelated code changes to the topic

**Review by simonguo — CHANGES_REQUESTED** [review]
You should add tests for the changes to Form and FormControl.
https://github.com/rsuite/rsuite/blob/main/CONTRIBUTING.md#testing

**Comment by giacomoaugello1** [comment]
Hi Simon,

I will make these changes, but just one more question...

How should I handle the "Unrelated code changes to the topic" warnings?
I edited that code only because the "npm run lint" command was throwing errors, preventing me from pushing the code. However, I can restore those files and force the push if you wish.

I'll let you know as soon as I make the changes.

Best regards,
Giacomo Augello.

**Comment by simonguo** [comment]
Hi @giacomoaugello1 

> How should I handle the "Unrelated code changes to the topic" warnings?

 You should revert these changes, don't change it.

> I edited that code only because the "npm run lint" command was throwing errors, preventing me from pushing the code. However, I can restore those files and force the push if you wish.

All submitted code must pass the "npm run lint" and "npm run test" checks. If the code does not pass the check, it will not be merged into the main branch. The reason your submitted code has not been checked is that you have downgraded the version of typescript from 4.6.4 to 4.4.3, which caused type export check errors.

I have reviewed your code. Please fix all the problematic code and add test code to resubmit.

**Comment by giacomoaugello1** [comment]
Hi Simon,

It's been a while, but I've been quite busy lately. I've added the tests to the schema-typed repository.

Once they're merged, I'll proceed to do the same with the rsuite repository.

Best regards,

Giacomo Augello

**Pull request closed without merging**